lib/vdsm/virt/vmdevices/storage.py

```diff
@@ -923,22 +925,54 @@ def get_snapshot_xml(self, snap_info):
 
     def on_block_threshold(self, reported_path):
         """
-        Callback to be executed when we receive a BLOCK_THRESHOLD
-        event from libvirt. We mark this drive accordingly,
-        so the periodic check will pick up this drive for
-        extension.
+        Called when disk allocation excceeds configured threshold.
+
+        Mark this drive for extension, to be picked by the periodic monitor
+        later.
+
+        Return True if threshold state was modified.
         """
         with self._lock:
             if reported_path != self._path:
                 self.log.debug(
                     "block threshold event mismatch drive %r path=%r "
                     "reported path=%r - ignored",
                     self.name, self._path, reported_path)
-                return
+                return False
+
+            if self._threshold_state == BLOCK_THRESHOLD.EXCEEDED:
+                self.log.debug(
+                    "drive %r block threshold already exceeded, ignored",
```

Comment:
You use here `%r`, why not use it also at the logs above, where you used `%s`? IMHO `%r` makes log more readable.

Reply:
Using %r for consistency with previous log (line 941), but I think we usually use %s for the drive name.

I'l leave this to a global log cleanup in this module when we finish this change.